pkg/services/notification/handler.go

```diff
@@ -75,83 +79,187 @@ func (s *Server) CreateNotification(ctx context.Context, req *pb.CreateNotificat
 	if err != nil {
 		return nil, err
 	}
+	logger.Debugf(ctx, "Create Notification[%s] in DB successfully.", notification.NotificationId)
 
-	tasks, err := SplitNotificationIntoTasks(notification)
+	_, err = s.createTasksByNotification(ctx, notification)
+	if err != nil {
+		logger.Errorf(ctx, "create Tasks By Notification failed, [%+v].", err)
+		return nil, err
+	}
+
+	// Enqueue notification after create tasks.
+	err = s.controller.notificationQueue.Enqueue(notification.NotificationId)
 	if err != nil {
-		logger.Errorf(ctx, "Split notification into tasks failed, [%+v]", err)
+		logger.Errorf(ctx, "Push notification[%s] into etcd failed, [%+v].", notification.NotificationId, err)
 		return nil, err
 	}
+	logger.Debugf(ctx, "Push notification[%s] into etcd successfully.", notification.NotificationId)
+	logger.Debugf(ctx, "Create Notification[%s] successfully.", notification.NotificationId)
 
+	return &pb.CreateNotificationResponse{
+		NotificationId: pbutil.ToProtoString(notification.NotificationId),
+	}, nil
+}
+
+func (s *Server) createTasksByNotification(ctx context.Context, nf *models.Notification) ([]*models.Task, error) {
+	tasks, err := rs.SplitNotificationIntoTasks(nf)
+	if err != nil {
+		logger.Errorf(ctx, "Split notification into tasks failed, [%+v].", err)
+		return nil, err
+	}
+
+	tasks, err = s.createTasks(ctx, tasks)
+	if err != nil {
+		logger.Errorf(ctx, "Split notification into tasks failed, [%+v].", err)
+		return nil, err
+	}
+	return tasks, nil
+}
+
+func (s *Server) createTasks(ctx context.Context, tasks []*models.Task) ([]*models.Task, error) {
+	var err error
 	for _, task := range tasks {
 		err = rs.RegisterTask(ctx, task)
 		if err != nil {
 			return nil, err
 		}
+		logger.Debugf(ctx, "Create Task[%s] in DB successfully.", task.TaskId)
+
 		err = s.controller.taskQueue.Enqueue(task.TaskId)
 		if err != nil {
-			logger.Errorf(ctx, "Push task [%s] into etcd failed, [%+v]", task.TaskId, err)
+			logger.Errorf(ctx, "Push task[%s] into etcd failed, [%+v].", task.TaskId, err)
 			return nil, err
 		}
+		logger.Debugf(ctx, "Push task[%s] into etcd successfully.", task.TaskId)
+		logger.Debugf(ctx, "Create Task[%s] successfully.", task.TaskId)
 	}
 
-	// Enqueue notification after tasks.
-	err = s.controller.notificationQueue.Enqueue(notification.NotificationId)
-	if err != nil {
-		logger.Errorf(ctx, "Push notification [%s] into etcd failed, [%+v]", notification.NotificationId, err)
-		return nil, err
-	}
-
-	return &pb.CreateNotificationResponse{
-		NotificationId: pbutil.ToProtoString(notification.NotificationId),
-	}, nil
+	return tasks, nil
 }
 
 func (s *Server) DescribeNotifications(ctx context.Context, req *pb.DescribeNotificationsRequest) (*pb.DescribeNotificationsResponse, error) {
 	nfs, nfCnt, err := rs.DescribeNotifications(ctx, req)
 	if err != nil {
-		logger.Errorf(ctx, "Failed to describe notifications, error: %+v", err)
+		logger.Errorf(ctx, "Failed to Describe Notifications, [%+v], [%+v].", req, err)
```

Comment:
No need to log req, the req will be log in grpc_server.

Reply:
updated.